test/API/H5_api_misc_test.c

```diff
@@ -1033,26 +1036,30 @@ test_symbols_in_compound_field_name(void)
     }
     H5E_END_TRY
 
-    return 1;
+    return;
 }
 
-int
-H5_api_misc_test(void)
+void
+H5_api_misc_test_add(void)
 {
-    size_t i;
-    int    nerrors;
-
-    printf("**********************************************\n");
-    printf("*                                            *\n");
-    printf("*          API Miscellaneous Tests           *\n");
-    printf("*                                            *\n");
-    printf("**********************************************\n\n");
-
-    for (i = 0, nerrors = 0; i < ARRAY_LENGTH(misc_tests); i++) {
-        nerrors += (*misc_tests[i])() ? 1 : 0;
-    }
-
-    printf("\n");
-
-    return nerrors;
+    int64_t testframe_flags = 0;
+
+    if (GetTestMaxNumThreads() > 1)
```

Comment:
I like the direction this solution has moved, but determining the test framework flags for a test at runtime like this is problematic because the testing framework doesn't parse command-line arguments until **after** all the tests have already been added, so this won't work in the expected way. Any changed value for `GetTestMaxNumThreads()` won't have been set yet. It essentially has to be baked in at compile-time whether a test is going to manage its own threads, or desires for the framework to handle them.

Reply:
I hadn't tested skipping tests, so I didn't catch that the order of operations was important here. It should be relatively painless to replace this flag with a similar one that delegates thread control to the framework, which can decide the number of threads at runtime based on the argument.